// karma/bin/test-main.js
require.config({
  // Karma serves files under /base, which is the basePath from your config file
  baseUrl: 'base/js',

  // dynamically load all test files
  deps: allTestFiles,

  // we have to kickoff jasmine, as it is asynchronous
  callback: window.__karma__.start
});

var tests = [];
for (var file in window.__karma__.files) {
  if (window.__karma__.files.hasOwnProperty(file)) {
    if (/Spec\.js$/.test(file)) {
      tests.push(file);
    }
  }
}

requirejs.config({
    // Karma serves files from '/base'
    baseUrl: 'base/js',

    paths: {
        jQuery: 'base/external/jquery-1.10.2.min',
        lodash: 'base/external/lodash',
        general: 'base/general',
        angular: 'base/external/angular.min',
        angularRoute: 'base/external/angular-route.min',
        mlcApp: 'base/mlc.app',

        translation: 'base/translation',
        en: 'base/locales/en',
        fr: 'base/locales/fr'
    },
    shim: {
        // Libraries
        'jQuery': {
            exports: '$'
        },
        'lodash': {
            exports: '_'
        },
        'angular': {
            exports: 'angular'
        },
        'angularRoute': {
            deps: ['angular']
        },
        'translation': {
            exports: 'translation'
        },
        'mlcApp': {
            deps: [
                'lodash',
                'jQuery',
                'angular',
                'angularRoute',
                'translation', 
                'general'
            ],
            exports: 'mlcApp'
        },

        // Local files
        'en': {
            deps: ['translation'],
            exports: 'en'
        },
        'fr': {
            deps: ['translation'],
            exports: 'fr'
        }
    },

    // ask Require.js to load these files (all our tests)
    deps: tests,

    // start test run, once Require.js is done
    callback: window.__karma__.start
});
